ding/entry/application_entry.py

```diff
@@ -137,10 +138,16 @@ def collect_demo_data(
     policy.collect_mode.load_state_dict(state_dict)
     collector = SampleSerialCollector(cfg.policy.collect.collector, collector_env, collect_demo_policy)
 
+    policy_kwargs = None if not hasattr(cfg.policy.other.get('eps', None), 'collect') \
+        else {'eps': cfg.policy.other.collect}
+
     # Let's collect some expert demostrations
-    exp_data = collector.collect(n_sample=collect_count)
+    exp_data = collector.collect(n_sample=collect_count, policy_kwargs=policy_kwargs)
     if cfg.policy.cuda:
         exp_data = to_device(exp_data, 'cpu')
-    with open(expert_data_path, 'wb') as f:
-        pickle.dump(exp_data, f)
+    if cfg.policy.learn.get('data_type', None) == 'hdf5':
```

Comment:
It would be better to place `data_type` in `policy.collect`. And its name can be `offline_data_save_type`.